code/game/objects/items/clay_items.dm

```diff
@@ -185,14 +185,28 @@
 	if (isliving(A) || isturf(A) || (isobj(A) && A.density))
 		shatter()
 
+/// This code is not to be confused with apply_hit_effect()
+///
+/// This code handles the throwing of bottles.
+///
 /obj/item/weapon/reagent_containers/food/drinks/clay/throw_impact(atom/hit_atom, var/speed)
 
 	..()
-	if (reagents)
+	if (reagents && reagents.total_volume > 0)
 		hit_atom.visible_message("<span class='notice'>The contents of \the [src] splash all over [hit_atom]!</span>")
 		reagents.splash(hit_atom, reagents.total_volume)
-	shatter(loc, hit_atom)
+		shatter(loc, hit_atom)
```

Comment:
No need to call `shatter(loc, hit_atom)`twice, just remove this instance on line 198 and remove an indentation at line 201.